routes/taspr.py

```diff
@@ -442,21 +484,33 @@ def create_csv(packaged_data):
     return output.getvalue()
 
 
-def return_csv(csv_data):
+def return_csv(csv_data, var_ep, place_id, lat=None, lon=None):
     """Return the CSV data as a download
 
     Args:
         csv_data (?): csv data created with create_csv() function
+        var_ep (str): tas, pr, or taspr
+        place (str): place name unless just a lat/lon value
+        lat: latitude unless an area
+        lon: longitude unless an area
 
     Returns:
         CSV Response
     """
+
+    place_name, place_type = place_name_and_type(place_id)
+
+    if place_name is not None:
```

Comment:
Maaaaaaybe we make a ticket to pull all of this related logic into a single place, since I see similar fragments throughout this PR.  Not sure.